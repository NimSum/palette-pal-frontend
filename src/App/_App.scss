@import '../variables.scss';

.App {
  text-align: center;
  position: relative;
}

.main {
  min-height: calc(100vh - 102px);
  position: relative;
}

.active {
  opacity: 1;
  font-weight: 600;
  border-bottom: 2px solid $btn-color;
}