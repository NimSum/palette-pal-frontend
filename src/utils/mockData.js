const mockData = {
  mockDetailedProjects: [
    {
      "project_name": "Project ONE",
      "project_id": 1,
      "palette_name": "Palette ONE",
      "palette_id": 1,
      "color_1": "#000000",
      "color_2": "#000000",
      "color_3": "#319547",
      "color_4": "#000000",
      "color_5": "#834267"
    },
    {
      "project_name": "Project TWO",
      "project_id": 2,
      "palette_name": "Palette TWO",
      "palette_id": 2,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267"
    },
    {
      "project_name": "Project ONE",
      "project_id": 2,
      "palette_name": "Palette THREE",
      "palette_id": 3,
      "color_1": "#123456",
      "color_2": "#654321",
      "color_3": "#245216",
      "color_4": "#784895",
      "color_5": "#235686"
    }
  ],
  mockProjects: [
    {
      "id": 1,
      "project_name": "Project ONE",
      "created_at": "2019-07-05T21:39:51.436Z",
      "updated_at": "2019-07-05T21:39:51.436Z"
    },
    {
      "id": 2,
      "project_name": "Project TWO",
      "created_at": "2019-07-05T21:40:51.436Z",
      "updated_at": "2019-07-05T21:40:51.436Z"
    }
  ],
  mockPalettes: [
    {
      "id": 1,
      "palette_name": "Test ONE",
      "project_id": 1,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267",
      "created_at": "2019-07-05T21:39:57.227Z",
      "updated_at": "2019-07-05T21:39:57.227Z"
    },
    {
        "id": 2,
        "palette_name": "Test TWO",
        "project_id": 2,
        "color_1": "#000000",
        "color_2": "#000000",
        "color_3": "#319547",
        "color_4": "#000000",
        "color_5": "#834267",
        "created_at": "2019-07-06T01:24:45.881Z",
        "updated_at": "2019-07-06T01:24:45.881Z"
    }
  ]
}

export default mockData;