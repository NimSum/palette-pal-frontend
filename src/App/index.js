import React from 'react';
import './_App.scss';
import Header from '../Header';
import SubHeader from '../SubHeader';
import { Switch, Route } from 'react-router-dom';

function App() {
  return (
    <div className="App">
      <Header />
      <SubHeader />
      <main>
        <Switch>
          {/* <Route exact path="/" component={PickerScreen} /> */}
          {/* <Route exact path="/" component={ProjectScreen} /> */}
          {/* <Route render={ErrorScreen} /> */}
        </Switch>
      </main>
    </div>
  );
}

export default App;
