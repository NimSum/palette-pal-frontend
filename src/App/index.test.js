import React from 'react';
import App from './index';
import { shallow, mount } from 'enzyme';
import requests from '../utils/apiRequests';
import mockData from '../utils/mockData';
import cleanCombinedData from '../utils/cleaners';

jest.mock('../utils/apiRequests');
jest.mock('../utils/cleaners');

requests.postNewUser.mockImplementation(() => 
  Promise.resolve(1));
requests.getDetailedProjects.mockImplementation(() => 
  Promise.resolve(mockData.mockDetailedProjects));
requests.loginUser.mockImplementation(() => 
  Promise.resolve({ token: 'VALID TOKEN', projects: mockData.mockDetailedProjects }))
cleanCombinedData.mockImplementation(() => mockData.mockCleanedProject);

describe('App', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<App />);
    instance = wrapper.instance();

    jest.spyOn(instance, 'checkForLogin');
  })

  it('should match component App snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call checkForLogin method when it mounts', () => {
    instance.componentDidMount();

    expect(instance.checkForLogin).toHaveBeenCalled()
  })

  it('should have the expected default state', () => {
    const defaultUserData = [];
    wrapper.setState({ userData: defaultUserData });
    expect(wrapper.state()).toEqual({
      authorized: false,
      projectData: [],
      userData: [],
      loading: false,
      err: '',
      showAcctDialog: false
    })
  })

  describe('User Validation', () => {
    let localStorageMock = { 
      getItem: () => JSON.stringify('true'),
      setItem: jest.fn()
    };
    Object.defineProperty(window, 'localStorage', { value: localStorageMock });
    
    it('should checkForLogin', async () => {
      await instance.checkForLogin();
      expect(instance.state.userData).toEqual(mockData.mockCleanedProject)
    })
  
    it('should logUserIn', async () => {
      const mockUser = {
        "email": "nimsum@nim.com",
        "password": "nimsum"
      };
      const expectedKey = 'user_token';
      const expectedToken = { token: "VALID TOKEN" };

      const result = await instance.logUserIn(mockUser);
      expect(result.token).toEqual(expectedToken.token);
      expect(localStorageMock.setItem).toHaveBeenCalledWith(expectedKey, JSON.stringify("VALID TOKEN"));
    })

    it('should signUserUp', async () => {
      const mockNewUser = {
        "email": "nimsum@nim.com",
        "password": "nimsum",
        "user_name": "nim"
      };

      await instance.signUserUp(mockNewUser);
      expect(requests.postNewUser).toHaveBeenCalledWith(mockNewUser);
    })

    it('should logUserOut', async () => {
      await instance.logUserOut();
      expect(localStorageMock.setItem).toHaveBeenCalledWith('user_token', "\"\"")
      expect(wrapper.state().userData).toEqual([]);
    })
  })
  
  describe.skip('updateProjectData', () => {
    const mockProject = mockData.mockProjects[1];

    it("should trigger post request for 'add' action", async () => {

    })

    it("should trigger delete request for 'delete' action", async () => {

    })

    it("should trigger put request for 'update' action", async () => {

    })

    it("should setState new userData after after api requests completion", async () => {

    })
    
  })
 
  describe.skip('updatePaletteData', () => {
    const mockPalette = mockData.mockPalettes[1];

    it("should trigger post request for 'add' action", async () => {

    })

    it("should trigger delete request for 'delete' action", async () => {

    })

    it("should trigger put request for 'update' action", async () => {

    })

    it("should setState new userData after after api requests completion", async () => {

    })
  })
})