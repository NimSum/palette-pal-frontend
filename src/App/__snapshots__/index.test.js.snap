// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match component App snapshot 1`] = `
<div
  className="loading-screen"
>
  <img
    alt="Loading icon"
    src="https://66.media.tumblr.com/09dc11b8b4b4e1be71dba1c570882308/tumblr_naksdbfjZp1sa11jco1_500.gif"
  />
  <h2>
    Loading...
  </h2>
</div>
`;
